File: frontend/src/api/audit.ts
import { apiGet, apiUrl, buildQuery } from "./client";

export interface AuditEntry {
  ts: string;
  event: string;
  outcome: string;
  user: string;
  detail: string;
}

export interface AuditPage {
  items: AuditEntry[];
  total_matching: number;
  total_available: number;
}

export interface AuditFilters {
  event?: string[];
  user?: string[];
  outcome?: string[];
  date_from?: string;
  date_to?: string;
}

export function fetchAudit(filters: AuditFilters = {}): Promise<AuditPage> {
  return apiGet<AuditPage>(`/audit${buildQuery(filters)}`);
}

export function auditExportCsvUrl(filters: AuditFilters): string {
  return apiUrl(`/audit/export.csv${buildQuery(filters)}`);
}

File: api/schemas/audit.py
# -*- coding: utf-8 -*-
from __future__ import annotations

from typing import List

from pydantic import BaseModel


class AuditEntryOut(BaseModel):
    ts: str
    event: str
    outcome: str
    user: str
    detail: str


class AuditPageOut(BaseModel):
    items: List[AuditEntryOut]
    total_matching: int
    total_available: int
